docs/architecture-decisions/adr006-avoid-react-fc.md

```diff
@@ -0,0 +1,42 @@
+# ADR006: Avoid React.FC and React.SFC
+
+## Context
+
+Facebook has removed ```React.FC``` from their base template for a Typescript project. The reason for this was that it was found to be an unnecessary feature with next to no benefits in combination with a few downsides.
+
+The main reasons were:
+- **children props** were implicitly added
+- **Generic Type** lacked support
```

Comment:
"Generic Types were not supported on children" may be a bit cleared

Reply:
I think generic types were not supported at all, e.g. if you want to infer type of something inside the component based on prop type, you can't do it with React.FC

Reply:
Ah OK! Then disregard!

Reply:
Sure! 👍